### DISCUSSION TITLE
What is required before pyHanko is production ready

### DISCUSSION THREAD

**Discussion by mikeylockwood** [MatthiasValvekens/pyHanko#341] [Q&A] [answered]
I have been trying out digital signing pdfs using python and pyHanko is by far the best package I've used so far. I see it has been moved from alpha to beta this year which is great to see. I was just wondering what is needed to be done before it is considered production ready. Is it what is documented on the known issues page or is there more? I'm just trying to get an idea if I should be using it in it's current form. Thanks!

**Comment by MatthiasValvekens** [comment, marked as answer]
There are a fair number of companies using pyHanko in production for signing PDFs already, and there have been for a while. :)

The chief reason why I don't want to move beyond beta just yet is because the validation API is still too much in flux. The validation API sees less production use, but I personally consider it quite important, so while I _could_ take the lazy way out, slap an "experimental" label on the whole thing and ship v1.0.0, I don't feel comfortable with that at this stage.

Conclusion: if signing is all you care about, things have been (more or less) stable for a long while. If you want to integrate pyHanko into a validation tool, then you'll possibly face a higher maintenance burden given that that API will still continue to evolve.

**Reply by mikeylockwood** [reply]
This is very helpful, thanks a lot :)
